roles/docker/templates/docker-svc.j2

```diff
@@ -1,17 +1,27 @@
 [Unit]
 Description=Docker Application Container Engine
 Documentation=https://docs.docker.com
-After=network.target docker.socket
-Requires=docker.socket
+After=network.target
 
 [Service]
 TimeoutStartSec=5m
 Type=notify
 {% if docker_device != "" %}
-ExecStart=/usr/bin/docker daemon --storage-opt dm.thinpooldev=/dev/mapper/contiv-dockerthin -H fd:// --cluster-store=etcd://localhost:{{ etcd_client_port1 }}
+{% set docker_devmapper_opts = "--storage-opt dm.thinpooldev=/dev/mapper/contiv-dockerthin" %}
 {% else %}
-ExecStart=/usr/bin/docker daemon -H fd:// --cluster-store=etcd://localhost:{{ etcd_client_port1 }}
+{% set docker_devmapper_opts = "" %}
 {% endif %}
+
+{% if docker_version | version_compare('1.12', '>=') %}
+{% set docker_daemon_startup = "dockerd" %}
+{% else %}
+{% set docker_daemon_startup = "docker daemon" %}
```

Comment:
@neelimamukiri: This was mostly a matter of how the systemd units were set up. They weren't absolutely necessary to run Docker. I've been running Docker for a long time without that setup.